src/caret_analyze/architecture/architecture.py

```diff
@@ -719,12 +729,26 @@ def update_message_context(self, context_type: str,
                                    'subscription_topic_name': subscribe_topic_name,
                                    'publisher_topic_name': publish_topic_name})
 
-    def remove_callback_chain(self, subscribe_topic_name: str, publish_topic_name: str) -> None:
+    def remove_callback_chain(
+        self,
+        subscribe_topic_name: str,
+        subscription_construction_order: int | None,
+        publish_topic_name: str,
+        publisher_construction_order: int
+    ) -> None:
         self._contexts = [
             context for context in self._contexts
-            if (context['subscription_topic_name'], context['publisher_topic_name'],
-                context['context_type']) !=
-               (subscribe_topic_name, publish_topic_name, 'callback_chain')
+            if (context['subscription_topic_name'],
+                0 if 'subscription_construction_order' not in context else
+                context['subscription_construction_order'],
+                context['publisher_topic_name'],
+                0 if 'publisher_construction_order' not in context else
+                context['publisher_construction_order'],
+                context['context_type']) != (subscribe_topic_name,
+                                             subscription_construction_order,
+                                             publish_topic_name,
+                                             publisher_construction_order,
+                                             'callback_chain')
```

Comment:
Sorry for repeated comments for same lines. I suggest using `dict.get()` to improve readability.

```suggestion
            if (context['subscription_topic_name'],
                context.get('subscription_construction_order', 0),
                context['publisher_topic_name'],
                context.get('publisher_construction_order', 0),
                context['context_type']) != (subscribe_topic_name,
                                             subscription_construction_order,
                                             publish_topic_name,
                                             publisher_construction_order,
                                             'callback_chain')
```